selftests/unit/test_archive.py

```diff
@@ -123,6 +124,17 @@ def get_path(*args):
         self.assertEqual(act, exp)
         self.assertEqual(os.path.realpath(get_path("link_to_dir")),
                          get_path("dir"))
+        # File permissions
+        self.assertEqual(os.stat(get_path("dir", "file2")).st_mode & 0o777,
```

Comment:
I like symbolic names where it makes sense, but I don't see a problem in using the orred value instead of constructing it all the time. The `0o777` is IMO easier to read than the chunk of stats...